src/tribler/core/utilities/pony_utils.py

```diff
@@ -28,6 +34,87 @@
 StatDict = Dict[Optional[str], core.QueryStat]
 
 
+class DatabaseIsCorrupted(Exception):
+    pass
+
+
+def table_exists(cursor: sqlite3.Cursor, table_name: str) -> bool:
+    cursor.execute("SELECT name FROM sqlite_master WHERE type='table' AND name=?", (table_name,))
+    return cursor.fetchone() is not None
+
+
+def get_db_version(db_path, default: int = None) -> int:
+    handle_db_if_corrupted(db_path)
+    if not db_path.exists():
+        version = None
+    else:
+        with marking_corrupted_db(db_path):
+            with contextlib.closing(sqlite3.connect(db_path)) as connection:
+                with connection:
+                    cursor = connection.cursor()
+                    if not table_exists(cursor, 'MiscData'):
+                        version = None
+                    else:
+                        cursor.execute("SELECT value FROM MiscData WHERE name == 'db_version'")
+                        row = cursor.fetchone()
+                        version = int(row[0]) if row else None
+
+    if version is not None:
+        return version
+
+    if default is not None:
+        return default
+
+    raise RuntimeError(f'The version value is not found in database {db_path}')
```

Comment:
I'd like to keep the previous version of the code. Otherwise, it may be hard to read it because the reader needs to take into account, for example, that version can theoretically be 0 or None, as well as the default value and the exact logic of the expression `(value or default) is not None` is not obvious, you need to consider how `0 or None` or `None or 0` are calculated in Python, etc. The logic in the PR is simple and does not require any complicated analysis.